modules/rostests/kmtests/rtl/RtlStrSafe.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * PROJECT:         ReactOS kernel-mode tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Test for ntstrsafe.h functions
+ * COPYRIGHT:       Copyright 2018 Hern�n Di Pietro <hernan.di.pietro@gmail.com>
+ *                  Copyright 2019 Colin Finck <colin@reactos.org>
+ */
+
+#define KMT_EMULATE_KERNEL
+#include <kmt_test.h>
+#include <ntstrsafe.h>
+#include <ntdef.h>
+#include <ndk/rtlfuncs.h>
+
+#define TESTAPI static void
+
+TESTAPI
+Test_RtlUnicodeStringPrintf()
+{
+    WCHAR Buffer[1024];
+    WCHAR OvrBuffer[1024];
+    WCHAR BufferSmall[16];
+    UNICODE_STRING UsString;
+    const WCHAR FormatStringInts[] = L"%d %d %d";
+    const WCHAR FormatStringStrs[] = L"%s %s %s";
+    const WCHAR Result[] = L"1 2 3";
+    UNICODE_STRING UsStringNull;
+    int i;
+
+    /* No zeros (Don't assume UNICODE_STRINGS are NULL terminated) */
+
+    RtlFillMemory(Buffer, sizeof(Buffer), 0xAA);
+    RtlFillMemory(BufferSmall, sizeof(BufferSmall), 0xAA);
+
+    /* STATUS_SUCCESS test */
+
+    UsString.Buffer = Buffer;
+    UsString.Length = 0;
+    UsString.MaximumLength = sizeof(Buffer);
+
+    KmtStartSeh();
+    ok_eq_hex(RtlUnicodeStringPrintf(&UsString, FormatStringInts, 1, 2, 3), STATUS_SUCCESS);
+    ok_eq_uint(UsString.Length, sizeof(Result) - sizeof(WCHAR));
+    ok_eq_uint(UsString.MaximumLength, sizeof(Buffer));
+    ok_eq_wchar(UsString.Buffer[0], L'1');
+    ok_eq_wchar(UsString.Buffer[1], L' ');
+    ok_eq_wchar(UsString.Buffer[2], L'2');
+    ok_eq_wchar(UsString.Buffer[3], L' ');
+    ok_eq_wchar(UsString.Buffer[4], L'3');
+    ok_eq_wchar(UsString.Buffer[5], L'\0');
+    
+    /* STATUS_BUFFER_OVERFLOW tests */
+
+    UsString.Buffer = BufferSmall;
+    UsString.Length = 0;
+    UsString.MaximumLength = 2;
+    
+    ok_eq_hex(RtlUnicodeStringPrintf(&UsString, FormatStringStrs, L"AAA", L"BBB", L"CCC"), STATUS_BUFFER_OVERFLOW);
+    ok_eq_uint(UsString.Length, UsString.MaximumLength);
+    ok_eq_wchar(UsString.Buffer[0], L'A');
+    ok_eq_wchar(UsString.Buffer[1], L'\0');
+    for (i = 2; i < sizeof(BufferSmall); i++)
+        ok_eq_wchar(UsString.Buffer[i], 0xAAAA);
+
+    UsString.Buffer = BufferSmall;
+    UsString.Length = 0;
+    UsString.MaximumLength = 7;
+
+    ok_eq_hex(RtlUnicodeStringPrintf(&UsString, FormatStringStrs, L"0123", L"4567", L"89AB"), STATUS_BUFFER_OVERFLOW);
+    ok_eq_uint(UsString.Length, UsString.MaximumLength);
+    ok_eq_wchar(UsString.Buffer[0], L'0');
```

Comment:
`wcscmp` would be fine here.

Reply:
Replaced all such code with `RtlCompareMemory` calls